include/deal.II/matrix_free/vector_data_exchange.h

```diff
@@ -247,6 +247,307 @@ namespace internal
         const std::shared_ptr<const Utilities::MPI::Partitioner> partitioner;
       };
 
+
+
+      /**
+       * Similar to the above but that the provided partitioner is pre-processed
+       * and indices of degrees of freedom that are shared are identified.
```

Comment:
I do not really understand this comment. Do you mean something along the lines
```suggestion
       * Similar to the above but using the internal data structures in the
       * partitioner in order to identify indices of degrees of freedom that are
       * in the same shared memory region.
```
And by `shared` you mean shared memory or something else? Please expand or adapt as appropriate.